src/lib.rs.in

```diff
@@ -140,31 +142,43 @@ macro_rules! es_body_op {
 
 impl Client {
     /// Create a new client
-    pub fn new(host: &str, port: u32) -> Client {
+    pub fn new(host: &str) -> Client {
+        let mut url = Url::parse(host).unwrap();
+
+        if url.port().is_none() {
```

Comment:
I'm wondering if it'd be better to leave the port exactly as specified in the url, and provide a separate function for a default host+port+protocol.